runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatboxTab.java

```diff
@@ -0,0 +1,101 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.chathistory;
+
+import com.google.common.collect.ImmutableList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import lombok.Getter;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.widgets.WidgetInfo;
+
+@Getter
+enum ChatboxTab
+{
+
+	ALL("All", "Switch tab", WidgetInfo.CHATBOX_TAB_ALL,
+		ChatMessageType.values()),
+
+	// null 'after' var since we're not adding to menu
+	PRIVATE("Private", null, WidgetInfo.CHATBOX_TAB_PRIVATE,
+		ChatMessageType.PRIVATECHAT, ChatMessageType.PRIVATECHATOUT, ChatMessageType.MODPRIVATECHAT,
+		ChatMessageType.LOGINLOGOUTNOTIFICATION),
+
+	// null 'after' var since we're not adding to menu
+	PUBLIC("Public", null, WidgetInfo.CHATBOX_TAB_PUBLIC,
+		ChatMessageType.PUBLICCHAT, ChatMessageType.AUTOTYPER, ChatMessageType.MODCHAT, ChatMessageType.MODAUTOTYPER),
+
+	GAME("Game", "Game: Filter", WidgetInfo.CHATBOX_TAB_GAME,
+		ChatMessageType.GAMEMESSAGE, ChatMessageType.ENGINE, ChatMessageType.BROADCAST,
+		ChatMessageType.SNAPSHOTFEEDBACK, ChatMessageType.ITEM_EXAMINE, ChatMessageType.NPC_EXAMINE,
+		ChatMessageType.OBJECT_EXAMINE, ChatMessageType.FRIENDNOTIFICATION, ChatMessageType.IGNORENOTIFICATION,
+		ChatMessageType.CONSOLE, ChatMessageType.SPAM, ChatMessageType.PLAYERRELATED, ChatMessageType.TENSECTIMEOUT,
+		ChatMessageType.WELCOME, ChatMessageType.UNKNOWN),
+
+	CLAN("Clan", "Clan: Off", WidgetInfo.CHATBOX_TAB_CLAN,
+		ChatMessageType.FRIENDSCHATNOTIFICATION, ChatMessageType.FRIENDSCHAT, ChatMessageType.CHALREQ_FRIENDSCHAT),
+
+	TRADE("Trade", "Trade: Off", WidgetInfo.CHATBOX_TAB_TRADE,
+		ChatMessageType.TRADE_SENT, ChatMessageType.TRADEREQ, ChatMessageType.TRADE, ChatMessageType.CHALREQ_TRADE),
+	;
+
+	private static final Map<Integer, ChatboxTab> TAB_MESSAGE_TYPES = new HashMap<>();
+
+	@Nullable
+	private final String after;
+	private final String name;
+	private final int widgetId;
+	private final List<ChatMessageType> messageTypes;
+
+	ChatboxTab(String name, String after, WidgetInfo widgetId, ChatMessageType... messageTypes)
+	{
+		this.name = name;
+		this.after = after;
+		this.widgetId = widgetId.getId();
+		this.messageTypes = ImmutableList.copyOf(messageTypes);
+	}
+
+	static
+	{
+		for (ChatboxTab t : values())
+		{
+			TAB_MESSAGE_TYPES.put(t.widgetId, t);
+		}
+	}
+
+	/**
+	 * Utility method that maps the type value to its respective
+	 * {@link ChatboxTab} value.
+	 *
+	 * @param widgetId packed widget ID
+	 * @return appropriate message types for a given chatbox tab widgetId,
+	 */
```

Comment:
The `@return` line is not at all accurate to what this method actually does. I don't think this method is complex enough to need javadoc, so this could be removed.